### ISSUE TITLE
Problem encoding VendorsConsent and VendorsLITransparency map

### ISSUE THREAD

**Issue by SachaMorard** [SirDataFR/iabtcfv2#1]
Hello
Thank you for this great library! 
I'm trying to encode a TC String but the generated string seems to be incomplete.
For example:
```golang
tcDataAcceptAll := &iabtcfv2.TCData{
                CoreString: &iabtcfv2.CoreString{
                    Version: 2,
                    Created: time.Now(),
                    LastUpdated: time.Now(),
                    CmpId: 123,
                    CmpVersion: 1,
                    ConsentScreen: 2,
                    ConsentLanguage: "FR",
                    VendorListVersion: 77,
                    TcfPolicyVersion: 2,
                    IsServiceSpecific: true,
                    UseNonStandardStacks: false,
                    SpecialFeatureOptIns: map[int]bool{
                        1: true,
                        2: true,
                    },
                    PurposesConsent: map[int]bool{
                        1: true,
                        2: true,
                    },
                    PurposesLITransparency: map[int]bool{
                        2: true,
                        3: true,
                    },
                    PurposeOneTreatment: false,
                    PublisherCC: "AA",
                    VendorsConsent: map[int]bool{
                        1: true,
                        2: true,
                        4: true,
                        6: true,
                    },
                    VendorsLITransparency: map[int]bool{
                        8: true,
                        11: true,
                        14: true,
                    },
                },
            }
            tcString := tcDataAcceptAll.ToTCString()
            fmt.Printf("%v", tcString)
```

It gives me:
`CPBxxAfPBxxAfB7ABCFRBNCsAMAAAGAAAAAAAAAAAAAA`

Then, if I decode this TC String, `vendorConsents` and `vendorLegitimateInterests` are empty...
```json
{
	"core": {
		"version": 2,
		"created": 1613595036700,
		"lastUpdated": 1613595036700,
		"cmpId": 123,
		"cmpVersion": 1,
		"consentScreen": 2,
		"consentLanguage": "FR",
		"vendorListVersion": 77,
		"policyVersion": 2,
		"isServiceSpecified": true,
		"useNonStandardStacks": false,
		"specialFeatureOptins": {
			"1": true,
			"2": true
		},
		"purposeConsents": {
			"1": true,
			"2": true
		},
		"purposeLegitimateInterests": {
			"2": true,
			"3": true
		},
		"purposeOneTreatment": false,
		"publisherCountryCode": "AA",
		"vendorConsents": {},
		"vendorLegitimateInterests": {},
		"publisherRestrictions": {}
	}
}
```

Do you know how to fix it?

**Comment by tla-sirdata** [comment]
Hi, thank you for using the lib and having raised this issue.

The `MaxVendorId`/`MaxVendorIdLI` fields should be defined for "non-range encoding" as their values can be different from the maximum values in `VendorConsents`/`VendorsLITransparency`.

For convenience, we added a fix to compute these values at encoding if they're not explicitly set.
Thus, when encoding the TC string (or the "Core String" segment), `MaxVendorId`/`MaxVendorIdLI` will be set with the maximum values found in `VendorConsents`/`VendorsLITransparency`.

The same logic has been applied to the `MaxVendorId` field in the "Disclosed Vendors" and "Allowed Vendors" segments.

The fix version is `v1.0.6`

**Comment by SachaMorard** [comment]
Amazing, thank you so much!
A bientôt
S
